packages/flutter_tools/lib/src/dart/generate_synthetic_packages.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:yaml/yaml.dart';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../build_system/build_system.dart';
+import '../build_system/targets/localizations.dart';
+
+Future<void> generateLocalizationsSyntheticPackage({
+  @required Environment environment,
+  @required BuildSystem buildSystem,
+}) async {
+  assert(environment != null);
+  assert(buildSystem != null);
+
+  final FileSystem fileSystem = environment.fileSystem;
+  final File l10nYamlFile = fileSystem.file(
+    fileSystem.path.join(environment.projectDir.path, 'l10n.yaml'));
+
+  // If pubspec.yaml has generate:true and if l10n.yaml exists in the
+  // root project directory, check to see if a synthetic package should
+  // be generated for gen_l10n.
+  if (l10nYamlFile.existsSync()) {
```

Comment:
you could drastically reduce the nesting here by re-arranging the code a bit:

```dart
if (!exists) {
  return;
}

if (condition that causes type error) {
  ...
  throwToolExit(..);
} 
```
i.e. check for errors first, and early return/throw